lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm

```diff
@@ -187,6 +188,28 @@ sub _validate_common_properties ($self) {
     $validation->optional('description');
 }
 
+# TODO: this exceed runtime limit for some reason
+sub _check_keep_logs_and_results ($self, $properties) {
+    if ($self->is_parent) {
+        if ( ($properties->{default_keep_logs_in_days} // 0) > ($properties->{default_keep_results_in_days} // 0)) {
+          $self->render(
+            json => {error => '`default_keep_logs_in_days` must be greater than `default_keep_results_in_days`'},
+            status => 400
+          );
+          return 1;
+        }
+    } else {
+        if ( ($properties->{keep_logs_in_days} // 0) > ($properties->{keep_results_in_days} // 0)) {
+          $self->render(
+            json => {error => '`keep_logs_in_days` must be greater than `keep_results_in_days`'},
+            status => 400
+          );
+          return 1;
+        }
+    }
+    return undef;
```

Comment:
TODO: why undef? 🤔

Reply:
That's correct style in Perl as the fallback.

Reply:
I guess you can also `return 0;`